src/java/map/ConcurrentFloatMask.java

```diff
@@ -97,18 +98,48 @@ public ConcurrentFloatMask clampMax(float value) {
         );
     }
 
+    public ConcurrentFloatMask clampMaxInArea(float value, ConcurrentBinaryMask area) {
+        return Pipeline.add(this, Arrays.asList(this, area), res ->
+                this.floatMask.clampMaxInArea(value, ((ConcurrentBinaryMask) res.get(1)).getBinaryMask())
+        );
+    }
+
     public ConcurrentFloatMask clampMin(float value) {
         return Pipeline.add(this, Collections.singletonList(this), res ->
                 this.floatMask.clampMin(value)
         );
     }
 
+    public ConcurrentFloatMask clampMinInArea(float value, ConcurrentBinaryMask area) {
+        return Pipeline.add(this, Arrays.asList(this, area), res ->
+                this.floatMask.clampMinInArea(value, ((ConcurrentBinaryMask) res.get(1)).getBinaryMask())
+        );
+    }
+
     public ConcurrentFloatMask threshold(float value) {
         return Pipeline.add(this, Collections.singletonList(this), res ->
                 this.floatMask.threshold(value)
         );
     }
 
+    public ConcurrentFloatMask useBrush(Vector2f location, String brushName, float intensity, int size) {
+        return Pipeline.add(this, Collections.singletonList(this), res ->
+                this.floatMask.useBrush(location, brushName, intensity, size)
+        );
+    }
+
+    public ConcurrentFloatMask useBrushRepeatedlyCenteredWithinArea(ConcurrentBinaryMask area, String brushName, int size, int frequency, float intensity) {
+        return Pipeline.add(this, Collections.singletonList(this), res ->
+                this.floatMask.useBrushRepeatedlyCenteredWithinArea(area.getBinaryMask(), brushName, size, frequency, intensity)
+        );
+    }
```

Comment:
This has the wrong dependencies list as area is not included. Also it needs to be res.get(1) not area in the function

Reply:
fixed